rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/metadata/CalculateDeltaNode.java

```diff
@@ -80,78 +86,34 @@ public void onMsg(TbContext ctx, TbMsg msg) {
             return;
         }
         JsonNode json = JacksonUtil.toJsonNode(msg.getData());
-        String inputKey = config.getInputValueKey();
         if (!json.has(inputKey)) {
```

Comment:
Possible only if JacksonUtil.toJsonNode(msg.getData()) will return null. This might happen only if msg.getData() is null. Is this possible in real cases? We never check for the msg.getData() is not null elsewhere.

Reply:
Fixed this by adding check for null